### PR TITLE
typo: remove extraneous "`" in doc comment, fix header

### PR DESCRIPTION
**Pull request by akesling**
## Which issue does this PR close?

N/A

## Rationale for this change

Improve documentation.

## What changes are included in this PR?

typo: remove extraneous "`" in doc comment

## Are these changes tested?

N/A

## Are there any user-facing changes?

No

### CODE REVIEW

File: datafusion/execution/src/runtime_env.rs
Comment (comphead) [inline datafusion/execution/src/runtime_env.rs:123]:
```suggestion
```

File: datafusion/execution/src/runtime_env.rs
Comment (comphead) [inline datafusion/execution/src/runtime_env.rs:121]:
```suggestion
    /// # Example: Register remote URL object store like [Github](https://github.com)
```

**Comment by comphead** [comment]
Would be that better WDYT @akesling ?

File: datafusion/execution/src/runtime_env.rs
Comment (akesling) [reply datafusion/execution/src/runtime_env.rs:121]: I'll defer to maintainers, but as a user I'd find that rather confusing given this is providing an example URL string as you would use in code, not a named service with associated link.

I wasn't sure if this was supposed to have a balanced "`" or not, so sent this as removing it (since the link already highlights the example).

**Comment by comphead** [comment]
Thanks @akesling I have changed the text, hope it is more clear now

**Review by alamb — APPROVED** [review]
Thank you @akesling  -- looks good to me

**Pull request merged**
